onnx/common/ir.h

```diff
@@ -866,6 +866,29 @@ friend struct Value;
 
   std::vector <OpSetID> opset_versions_;
 
+  bool isNameInAnyNode(const std::string& name) const {
+      auto f = [&name](const Value* v){ return v->uniqueName() == name; };
+      for(const Node* node : all_nodes) {
+          auto found_in = std::find_if(node->inputs().begin(), node->inputs().end(), f);
+          auto found_out = std::find_if(node->outputs().begin(), node->outputs().end(), f);
+          if(found_in != node->inputs().end() || found_out != node->outputs().end()) {
```

Comment:
nit: If you do the check for found_in before checking the node outputs then you can exit early when the name is found in node inputs.